File: CohesiveRP.Core.WebApi/ResponseDtos/BackgroundQueries/BackgroundQueryResponseDto.cs
using System.Net;
using System.Text.Json.Serialization;
using CohesiveRP.Common.WebApi;
using CohesiveRP.Storage.DataAccessLayer.BackgroundQueries.BusinessObjects;

namespace CohesiveRP.Core.WebApi.ResponseDtos.BackgroundQueries
{
    public class BackgroundQueryResponseDto : IWebApiResponseDto
    {
        [JsonConverter(typeof(JsonNumberEnumConverter<HttpStatusCode>))]
        [JsonPropertyName("code")]
        public HttpStatusCode HttpResultCode { get; set; }

        [JsonPropertyName("chatId")]
        public string ChatId { get; set; }

        [JsonPropertyName("linkedId")]
        public string LinkedId { get; set; }

        [JsonPropertyName("backgroundQueryId")]
        public string BackgroundQueryId { get; set; }

        [JsonPropertyName("dependenciesTags")]
        public List<string> DependenciesTags { get; set; }

        [JsonPropertyName("status")]
        public BackgroundQueryStatus Status { get; set; }

        [JsonPropertyName("priority")]
        public BackgroundQueryPriority Priority { get; set; }

        [JsonPropertyName("content")]
        public string Content { get; set; }

        [JsonPropertyName("tags")]
        public List<string> Tags { get; set; }
    }
}

File: CohesiveRP.UI.WebFrontend/src/ResponsesDto/chat/BackgroundQueryResponseDto.ts
import type { ServerApiResponseDto } from "../ServerApiResponseDto";

interface BackgroundQueryResponseDto extends ServerApiResponseDto {
    chatId : string,
    linkedId : string,
    backgroundQueryId : string,
    dependenciesTags : string[],
    tags : string[],
    status : string,
    priority : number,
    content : string,
    startFocusedGenerationDateTimeUtc: string,
    endFocusedGenerationDateTimeUtc: string,
    createdAtUtc : string
}

export type {
    BackgroundQueryResponseDto
};